### ISSUE TITLE
::placeholder not vendor prefixed in production

### ISSUE THREAD

**Issue by jaydenseric** [zeit/styled-jsx#305]
I don't know why, but `::placeholder` rules are not vendor prefixed in production builds:

<img width="892" alt="screen shot 2017-10-11 at 4 25 00 pm" src="https://user-images.githubusercontent.com/1754873/31423514-008fe3e4-aea1-11e7-81c0-c64691262659.png">

This results in no placeholder styles in Edge and IE.

Weirdly, prefixes appear in development:

<img width="874" alt="screen shot 2017-10-11 at 4 38 35 pm" src="https://user-images.githubusercontent.com/1754873/31423870-e5791b3c-aea2-11e7-822f-68e6c3160e7f.png">

Stylis is meant to handle this, so is it something going on with Next.js/styled-jsx?

<img width="651" alt="screen shot 2017-10-11 at 4 30 25 pm" src="https://user-images.githubusercontent.com/1754873/31423616-971fe9c6-aea1-11e7-9800-af7d151b5746.png">

**Comment by giuseppeg** [comment]
@thysultan could you help me out fixing this / improving this plugin? https://github.com/zeit/styled-jsx/blob/master/src/lib/style-transform.js#L68-L133 You know Stylis inside out so I think we could avoid a lot of bugs if you could fix / refactor that plugin.

**Comment by thysultan** [comment]
I think i have a fix for this, but it seems like `insertRule` throws on the placeholder vendor prefix. 

The [rule-sheet](https://github.com/thysultan/stylis.js/tree/master/plugins/rule-sheet) plugin makes use of this and you can see what throws when passed to `insertRule` in the [console](https://rawgit.com/thysultan/stylis.js/master/plugins/rule-sheet/tests/index.html)

**Comment by giuseppeg** [comment]
>  it seems like insertRule throws on the placeholder vendor prefix

That's fine I guess as soon as it won't throw in the target browser right? I mean `::-moz-placeholder` throws in Chrome and it is ignored (cool), doesn't throw in Firefox (cool).

**Comment by giuseppeg** [comment]
it might only be an issue when selectors are grouped probably eg.

```css
.foo,
::placeholder { color: red }
```

Since it throws `.foo` won't get styled

**Comment by thysultan** [comment]
Yes, i guess we could just not log in the catch block for errors that come from `placeholder` vendor prefixes.

The only other error is from `@supports (position:-webkit-sticky) or (position:sticky)`. Though i don't yet know if it's because the syntax is incorrect or because of the vendor prefix `(position:-webkit-sticky)`.

**Comment by giuseppeg** [comment]
I think because of the vendor prefix

**Comment by thysultan** [comment]
:smile: It's a syntax issue. The space after and before `or` is required.

This works.

```css
@supports (position:-webkit-sticky) or (position:sticky){.css-hash{background:red;}}
```

Will patch this in stylis.

**Comment by giuseppeg** [comment]
@thysultan so basically I could use your plugin and ditch mine?

**Comment by thysultan** [comment]
Patched in 3.3.2 and updated the [rule-sheet](https://github.com/thysultan/stylis.js/tree/master/plugins/rule-sheet) plugin. It should be a good working example of how to implement an insertRule plugin with the least overhead.

>so basically I could use your plugin and ditch mine

I think so.

**Comment by giuseppeg** [comment]
@jaydenseric I'll wait for @thysultan to fix his rule-sheet plugin hopefully I can switch to that and we can get continuous bug fixes and support. In the meantime if this is a real issue you can disable the `optimizedForSpeed` mode in production

**Comment by jaydenseric** [comment]
@giuseppeg thanks for the update, and tip.

**Comment by giuseppeg** [comment]
Fixed in 2.1.0
